web/client/src/components/infrastructure/infrastructure-map.vue

```diff
@@ -0,0 +1,284 @@
+	bearing: 0,
+};
+
+export default defineComponent({
+	name: 'InfrastructureMap',
+
+	data() {
+		return {
+			libreGLMap: null as (Map | null),
+			legendControlTypes,
+			initiallyCheckedControls,
+			iconUrl,
+			iconExtension,
+			elementTypeLabels: elementTypeLabels as { [elementType: string]: string },
+		};
+	},
+
+	computed: {
+		...mapState(InfrastructureNamespace, [
+			'currentInfrastructure',
+			'highlightedSignalStationRouteID',
+			'highlightedStationRouteID',
+		]),
+	},
+
+	watch: {
+		currentInfrastructure(newInfrastructure: string | null) {
+			// Re-instantiating the map on infrastructure change currently leads to duplicated icon fetching on change.
+			// @ts-ignore type instantiation for some reason is too deep
+			this.libreGLMap = newInfrastructure ? this.createMap(newInfrastructure) : null;
+		},
+
+		highlightedSignalStationRouteID(newID, oldID) {
+			if (!this.libreGLMap)
+				return;
+
+			if (newID) {
+				// @ts-ignore
+				highlightSignalStationRoute(this.libreGLMap, this.currentInfrastructure, newID);
+			} else {
+				// @ts-ignore
+				deHighlightSignalStationRoute(this.libreGLMap, oldID);
+			}
+		},
+
+		highlightedStationRouteID(newID, oldID) {
+			if (newID) {
+				// @ts-ignore
+				highlightStationRoute(this.libreGLMap, this.currentInfrastructure, newID);
+			} else {
+				// @ts-ignore
+				deHighlightStationRoute(this.libreGLMap, oldID);
+			}
+		},
+	},
+
+	methods: {
+		resize() {
+			if (!this.libreGLMap)
+				return;
+
+			this.libreGLMap.resize();
+		},
+
+		hasImage(elementType: string) {
+			return !specialLayoutControls.includes(elementType);
+		},
+
+		onLegendControlClicked(event: Event) {
+			if (specialLayoutControls.includes((event.target as HTMLInputElement).id)) {
+				this.evaluateSpecialLegendControls();
+
+				return;
+			}
+
+			this.evaluateLegendControlForControlType((event.target as HTMLInputElement).value);
+		},
+
+		evaluateLegendControlForControlType(type: string) {
+			if (!this.libreGLMap)
+				return;
+
+			// @ts-ignore
+			this.libreGLMap.setLayoutProperty(type + '-layer', 'visibility', this.$refs[type][0].checked ? 'visible' : 'none');
+
+			if (type !== 'station') {
+				// @ts-ignore
+				this.libreGLMap.setLayoutProperty('circle-' + type + '-layer', 'visibility', this.$refs[type][0].checked ? 'visible' : 'none');
+			}
+		},
+
+		evaluateSpecialLegendControls() {
+			if (!this.libreGLMap)  {
+				return;
+			}
+
+			const rising_checked = (this.$refs.Rising as HTMLInputElement).checked;
+			const falling_checked = (this.$refs.Falling as HTMLInputElement).checked;
+
+			let filter: FilterSpecification;
+			if (!rising_checked && falling_checked) {
+				filter = ['!', ['get', 'rising']];
+			} else if (rising_checked && !falling_checked) {
+				filter = ['get', 'rising'];
+			} else if (!rising_checked && !falling_checked) {
+				filter = ['boolean', false];
+			}
+
+			elementTypes.forEach((elementType) => {
+				if (elementType === 'station') {
+					return;
+				}
```

Comment:
Some early return have the brackets some not.

Reply:
We can maybe consider using a linter rule for this to make it consistent. What would be your preferred option? One can realize both "brackets everywhere" and "brackets only where needed" with the ESLint rule:
![image](https://user-images.githubusercontent.com/78490564/210009562-1d3b5e93-5411-426f-9c4b-6b29b6de3a73.png)

I would be okay with just enforcing brackets everywhere as they prevent bugs from forgetting to wrap the conditioned statement in brackets. Let me know your opinion on this.

Reply:
Yeah sounds good. Doesnt hurt to have them